components/Buttons/src/MDCButton.h

```diff
@@ -73,6 +73,24 @@
 @property(nonatomic) UIEdgeInsets hitAreaInsets;
 
 /**
+ The minimum size of the button’s alignment rect. If either the height or width are non-positive
+ (negative or zero), they will be ignored and that axis will adjust to its content size.
+
+ Defaults to CGSizeZero.
+ */
+@property(nonatomic, assign) CGSize minimumSize UI_APPEARANCE_SELECTOR;
+
+/**
+ The maximum size of the button’s alignment rect. If either the height or width are non-positive
+ (negative or zero), they will be ignored and that axis will adjust to its content size. Setting a
+ maximum size may result in image clipping or text truncation.
+
+ Defaults to CGSizeZero.
+ */
+@property(nonatomic, assign) CGSize maximumSize UI_APPEARANCE_SELECTOR;
```

Comment:
what happens if we set the max to smaller than the min? or vice versa

Reply:
Then undefined behavior will result.  I'm not sure that we should add logic to handle that situation.

Reply:
looks like flexible header does clamping
https://github.com/material-components/material-components-ios/blob/develop/components/FlexibleHeader/src/MDCFlexibleHeaderView.m#L1331

Reply:
We discussed this offline. I believe we agreed to keep the implementation as-is for now and add clamping if needed in the future.